frontend/src/metabase/querying/notebook/components/Notebook/use-run-visualization.ts

```diff
@@ -0,0 +1,52 @@
+import * as Lib from "metabase-lib";
+import type Question from "metabase-lib/v1/Question";
+
+type UseVisualizationProps = {
+  question?: Question;
+  isDirty: boolean;
+  isResultDirty: boolean;
+  updateQuestion: (question: Question) => Promise<void>;
+  runQuestionQuery: () => Promise<void>;
+  setQueryBuilderMode?: (mode: string) => void;
+};
+
+export const useRunVisualization = ({
+  question,
+  isDirty,
+  isResultDirty,
+  updateQuestion,
+  runQuestionQuery,
+  setQueryBuilderMode,
+}: UseVisualizationProps) => {
+  const cleanupQuestion = async () => {
```

Comment:
Nit: maybe this can be lifted out to be a simple util `function cleanQuestion(question: Question): Question`?

You could then replace the call site with:

```ts
if (isDirty && question) {
  await updateQuestion(cleanupQuestion(question))
}
```